src/mesh/BoundingBox.cpp

```diff
@@ -33,7 +33,7 @@ BoundingBox::BoundingBox(int dimension)
 
 bool BoundingBox::operator==(const BoundingBox &otherBB) const
 {
-  PRECICE_ASSERT(_dimensions == otherBB._dimensions, "Bounding boxes with different dimensions cannot be compared. Dimensions: " << _dimensions << " and " << otherBB._dimensions);
+  PRECICE_ASSERT(_dimensions == otherBB._dimensions, "Bounding boxes with different dimensions cannot be compared.", _dimensions, otherBB._dimensions);
```

Comment:
After https://github.com/precice/precice/pull/997#issuecomment-825541542 I think I understand the intention here. Formally there is a change in the output statement, because it does not say `Dimensions: 3 and 4` anymore. But I guess: Since it's an assertion that's what we want anyway and the statement before was just too verbose.